sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -50,170 +46,195 @@ namespace Azure { namespace Core {
-    }
+    static DateTime Now();
 
     /**
-     * @brief Construct an uninitialized (!@IsInitialized()) instance of @DateTime.
+     * @brief Construct an instance of @DateTime.
+     *
+     * @param year Year.
+     * @param month Month.
+     * @param day Day.
+     * @param hour Hour.
+     * @param minute Minute.
+     * @param second Seconds.
+     *
+     * @throw std::invalid_argument If any parameter is invalid.
      */
-    DateTime() : m_interval(0) {}
+    explicit DateTime(
+        int16_t year,
+        int8_t month = 1,
+        int8_t day = 1,
+        int8_t hour = 0,
+        int8_t minute = 0,
+        int8_t second = 0);
 
     /**
      * @brief Create @DateTime from a string representing time in UTC in the specified format.
      *
-     * @param timeString A string with the date and time.
-     * @param format A format to which /p timeString adheres to.
+     * @param dateTime A string with the date and time.
+     * @param format A format to which /p dateTime string adheres to.
      *
-     * @return @DateTime that was constructed from the \p timeString; Uninitialized
-     * (!@IsInitialized()) @DateTime if parsing \p timeString was not successful.
+     * @return @DateTime that was constructed from the \p dateTime string.
      *
-     * @throw DateTimeException If \p format is not recognized.
+     * @throw std::invalid_argument If \p format is not recognized, or if parsing error.
      */
-    static DateTime FromString(
-        std::string const& timeString,
-        DateFormat format = DateFormat::Rfc1123);
+    static DateTime Parse(std::string const& dateTime, DateFormat format);
 
   private:
     /**
      * @brief Get a string representation of the @DateTime.
      *
      * @param format The representation format to use.
      * @param fractionFormat The format for the fraction part of the Datetime. Only supported by
-     * ISO8601
+     * RFC 3339.
      *
-     * @throw DateTimeException If year exceeds 9999, or if \p format is not recognized.
+     * @throw std::length_error If year exceeds 9999.
+     * @throw std::invalid_argument If \p format is not recognized.
      */
-    std::string ToString(DateFormat format, TimeFractionFormat fractionFormat) const;
+    std::string GetString(DateFormat format, TimeFractionFormat fractionFormat) const;
 
   public:
     /**
      * @brief Get a string representation of the @DateTime.
      *
      * @param format The representation format to use.
      *
-     * @throw DateTimeException If year exceeds 9999, or if \p format is not recognized.
+     * @throw std::length_error If year exceeds 9999.
+     * @throw std::invalid_argument If \p format is not recognized.
      */
-    std::string ToString(DateFormat format = DateFormat::Rfc1123) const
+    std::string GetString(DateFormat format) const
     {
-      return ToString(format, TimeFractionFormat::DropTrailingZeros);
+      return GetString(format, TimeFractionFormat::DropTrailingZeros);
```

Comment:
Why GetString over ToString?

Reply:
This is what we decided during the API review. There was no clear winner, but we wanted consistency. Uuid had GetString(), so this one has the same name.

Reply:
We followed up in the chat, we will discuss wit the team whether all GetString()s should be ToString()s.